daft/catalog/__unity.py

```diff
@@ -0,0 +1,87 @@
+"""WARNING! These APIs are internal; please use Catalog.from_unity() and Table.from_unity()."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from daft.catalog import Catalog, Table
+from daft.unity_catalog import UnityCatalog as InnerCatalog  # noqa: TID253
+from daft.unity_catalog import UnityCatalogTable as InnerTable  # noqa: TID253
+
+if TYPE_CHECKING:
+    from daft.dataframe import DataFrame
+
+
+class UnityCatalog(Catalog):
+    _inner: InnerCatalog
+
+    def __init__(self, unity_catalog: InnerCatalog):
+        """DEPRECATED: Please use `Catalog.from_unity`."""
+        self._inner = unity_catalog
+
+    @staticmethod
+    def _try_from(obj: object) -> UnityCatalog | None:
+        """Returns an UnityCatalog instance if the given object can be adapted so."""
+        if isinstance(obj, InnerCatalog):
+            return UnityCatalog(obj)
+        return None
+
+    @property
+    def inner(self) -> InnerCatalog:
+        """Returns the inner unity catalog."""
+        return self._inner
+
+    ###
+    # get_*
+    ###
+
+    def get_table(self, name: str) -> UnityTable:
+        return UnityTable(self._inner.load_table(name))
+
+    ###
+    # list_.*
+    ###
+
+    def list_tables(self, pattern: str | None = None) -> list[str]:
+        if pattern is None or pattern == "":
+            return [
+                tbl
+                for cat in self._inner.list_catalogs()
+                for schema in self._inner.list_schemas(cat)
+                for tbl in self._inner.list_tables(schema)
+            ]
+        num_namespaces = pattern.count(".")
+        if num_namespaces == 0:
+            catalog_name = pattern
+            return [tbl for schema in self._inner.list_schemas(catalog_name) for tbl in self._inner.list_tables(schema)]
+        elif num_namespaces == 1:
+            schema_name = pattern
+            return [tbl for tbl in self._inner.list_tables(schema_name)]
+        else:
+            raise ValueError(
+                f"Unrecognized catalog name or schema name, expected a '.'-separated namespace but received: {pattern}"
+            )
+
+
+class UnityTable(Table):
+    _inner: InnerTable
+
+    def __init__(self, unity_table: InnerTable):
+        self._inner = unity_table
+
+    @staticmethod
+    def _try_from(obj: object) -> UnityTable | None:
```

Comment:
nit: I think `try_from... -> X | None` isn't very common in Python.

Instead, just throw `ValueError` if it isn't what's expected

Reply:
Ok, I added these for the permissive trying of different variants so specifically no raising. It doesn't make a functional difference so I can easily update this. I do think raising is safer in some way too .. it fails loudly

Reply:
Updated in latest revision.